### PR TITLE
Shoot candles with lasers to light them.

### PR DESCRIPTION
**Pull request by Repth1**
[tweak]
as title

shooting over them to light them would have been cool but i have no idea how to do that
For #21305

:cl:
 * rscadd: You can now shoot candles with lasers to light them.

### CODE REVIEW

**Comment by Goldsteel** [comment]
How about emitters, too? Does it work for them or are they not a beam projectile?

**Comment by Repth1** [comment]
emitters fire beam projectiles, but I'm pretty sure they only hit things that can block them, so it won't hit the candle by default

i'd have to figure out how to light the candle by having a beam projectile on the same turf instead

**Comment by Kitfox88** [comment]
Now let us light cigs and cigars with lasers, with a small chance to shoot yourself in the face instead.

File: code/game/objects/items/candle.dm [hunk 1]
```diff
@@ -63,6 +63,11 @@
 		update_icon()
 		set_light(0)
 
+/obj/item/candle/bullet_act(var/obj/item/projectile/Proj)
+	if(istype(Proj ,/obj/item/projectile/beam))
+		if(!istype(Proj ,/obj/item/projectile/beam/lasertag) && !istype(Proj ,/obj/item/projectile/beam/practice) && !Proj.nodamage)
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/game/objects/items/candle.dm]: Don't the lasertag and practice beams already have nodamage?

Comment (Repth1) [reply code/game/objects/items/candle.dm]: strangely enough, it doesn't seem like it, they just have 0 damage

Comment (Exxion) [reply code/game/objects/items/candle.dm]: nodamage is only used for stuns IIRC
But it seems easier to just check if damage is 0

Comment (Repth1) [reply code/game/objects/items/candle.dm]: makes sense, updated that check

**Comment by Shadowmech88** [comment]
Shooting over them may be possible, but you'd have to check for the existence of a candle on the current tile in whatever code is responsible for drawing the laser overlays, for every overlay, and I doubt that'd be resource efficient for lasers given how frequent and long they can be.

**Comment by Exxion** [comment]
Nah
`Crossed()`

**Comment by Shadowmech88** [comment]
For some reason I forgot that laser projectiles actually traveled at all.

**Pull request merged**